spec/whiteboard_http_server_test.go

```diff
@@ -0,0 +1,141 @@
+package spec
+
+import (
+	"bytes"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/pivotal-sydney/whiteboardbot/app"
+	. "github.com/pivotal-sydney/whiteboardbot/http"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strconv"
+)
+
+func makeRequest(params map[string]string) *http.Request {
+	data := url.Values{}
+
+	for k, v := range params {
+		data.Set(k, v)
+	}
+
+	request, _ := http.NewRequest("POST", "/", bytes.NewBufferString(data.Encode()))
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	request.Header.Add("Content-Length", strconv.Itoa(len(data.Encode())))
+	return request
+}
+
+var _ = Describe("WhiteboardHttpServer", func() {
+	var (
+		token          string
+		writer         *httptest.ResponseRecorder
+		mockWhiteBoard MockWhiteboard
+		handlerFunc    http.HandlerFunc
+		params         map[string]string
+		slackUser      SlackUser
+		slackChannel   SlackChannel
+	)
+
+	BeforeEach(func() {
+		token = os.Getenv("SLACK_TOKEN")
+
+		params = map[string]string{
+			"token":        "123",
+			"user_name":    "aleung",
+			"user_id":      "C987",
+			"channel_id":   "C456",
+			"channel_name": "sydney-standup",
+		}
+
+		slackUser = SlackUser{Username: "aleung", Author: "Andrew Leung", TimeZone: "Australia/Sydney"}
+		slackChannel = SlackChannel{Id: "C456", Name: "sydney-standup"}
+
+		os.Setenv("SLACK_TOKEN", "123")
+
+		slackClient := MockSlackClient{}
+		slackClient.AddSlackUser("C987", slackUser)
+
+		writer = httptest.NewRecorder()
+		mockWhiteBoard = MockWhiteboard{}
+		whiteboardServer := WhiteboardHttpServer{SlackClient: &slackClient, Whiteboard: &mockWhiteBoard}
+
+		handlerFunc = whiteboardServer.NewHandleRequest()
+	})
+
+	AfterEach(func() {
+		os.Setenv("SLACK_TOKEN", token)
```

Comment:
We discussed centralizing the reading of environment variables to a single place executed once during startup. The values read from the environment would be stored in a config object of sorts and made available wherever needed. That would make environment manipulation like this unnecessary.